test/functorch/test_control_flow.py

```diff
@@ -1576,5 +1594,55 @@ def forward(self, arg0_1: f32[s0, s1]):
             gm = make_fx(foo, tracing_mode='symbolic')(torch.ones(3, 4))
             self._expected_inline_normalized(gm.print_readable(print_output=False), exp_graph)
 
+    def _create_test_fns_for_cond(self, pred, inner_most_fn, operands, closure_list, nested_level):
+        if nested_level == 0:
+            if len(closure_list) > 0:
+                def true_fn(*operands):
+                    return inner_most_fn(*operands) + inner_most_fn(*closure_list)
+
+                def false_fn(*operands):
+                    return inner_most_fn(*operands) - inner_most_fn(*closure_list)
+            else:
+                def true_fn(*operands):
+                    return inner_most_fn(*operands)
+
+                def false_fn(*operands):
+                    return inner_most_fn(*operands)
+
+            def fn(pred, operands):
+                if len(operands) == 0 and len(closure_list) == 0:
+                    return torch.zeros(1)
+                return cond(pred, true_fn, false_fn, operands)
+            return (pred, operands), fn
+        else:
+            args, inner_fn = self._create_test_fns_for_cond(pred <= 0, inner_most_fn, operands, closure_list, nested_level - 1)
+
+            def true_fn(*operands):
+                return inner_most_fn(*operands) + inner_fn(*args)
+
+            def false_fn(*operands):
+                return inner_most_fn(*operands) - inner_fn(*args)
+
+            def fn(pred, operands):
+                if len(operands) == 0 and len(closure_list) == 0:
+                    return torch.ones(1)
+                return cond(pred, true_fn, false_fn, operands)
+            return (pred, operands), fn
+
+    @parametrize("pred", [True, torch.tensor(0), torch.randn(1), torch.tensor(False)])
+    @parametrize("inner_most_fn", [_reduce_func, _ReduceMod(), _ReduceObj()])
+    @parametrize("operands", [(), (torch.randn(2, 3), torch.randn(2, 3))])
+    @parametrize("closure_list", [[], [torch.randn(2, 3), torch.randn(2, 3)]])
+    @parametrize("nested_level", [0, 2])
+    def test_cond_tracing_with_valid_inputs(self, pred, inner_most_fn, operands, closure_list, nested_level):
```

Comment:
What does the tests print as? you might need to use subTest to get the printing to look better (it should not have any special characters in it)

Reply:
yes, it looks pretty bad.. Let me check how to use subTest.